### ISSUE TITLE
Collapsed regions expand by themselves

### ISSUE THREAD

**Issue by thany** [microsoft/vscode#88851]
Issue Type: <b>Bug</b>

Collapsed functions may expand by themselves at an arbitrary moment in time. I don't exactly know when or why this happens, but it always happens when I'm not looking at those functions, and find that I suddenly have to scroll much farther than I expected.

So far, this is happening in TSX files, but I wouldn't be surprized if it wasn't isolated to TSX.

VS Code version: Code 1.41.1 (26076a4de974ead31f97692a0d32f90d735645c0, 2019-12-18T14:58:56.166Z)
OS version: Windows_NT x64 10.0.18363

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz (12 x 2904)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: disabled_off<br>surface_control: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.74GB (14.44GB free)|
|Process Argv||
|Screen Reader|no|
|VM|67%|
</details><details><summary>Extensions (22)</summary>

Extension|Author (truncated)|Version
---|---|---
better-comments|aar|2.0.5
overtype|ada|0.2.0
Bookmarks|ale|10.6.0
color-info|bie|0.5.1
mustache|daw|1.1.1
vscode-eslint|dba|2.0.14
EditorConfig|Edi|0.14.4
vscode-great-icons|emm|2.1.47
file-icons|fil|1.0.21
duplicate|gee|1.0.1
docthis|joe|0.7.1
rainbow-csv|mec|1.4.0
vscode-duplicate|mrm|1.2.1
vscode-docker|ms-|0.9.0
remote-wsl|ms-|0.41.7
vscode-versionlens|pfl|0.24.0
vscode-template-literal-editor|pli|0.9.0
vscode-sort-json|ric|1.18.0
code-settings-sync|Sha|3.4.3
stylelint|shi|0.51.0
mdx|sil|0.1.0
vscode-svg-previewer|vit|0.5.1

(1 theme extensions excluded)

</details>

**Comment by aeschli** [comment]
Does it reproduce with extensions disabled?

**Comment by thany** [comment]
I don't even know what happens that reproduces it *with* extensions. Any suggestions?

**Comment by Stephen-Murphy** [comment]
Possibly related: I can repro a form of this with typescript and javascript files using the code below, I've had this issue happen many times without knowing what was causing it until I investigated today.

When the `bar` method is collapsed, and I press backspace a few times at the end of line 3, for a brief moment the template literal encapsulates the entire file, so everything after that line becomes part of the template, and all collapsed/folded regions after that point in the file are removed. If you don't see this happening and you close or remove the template string before scrolling down, you wouldn't know what had happened.

````typescript
export class Foo {
    foo() {
        return ``; // backspace
    }
    bar() { // collapse me
        return true;
    }
}
````

Becomes:

````typescript
export class Foo {
    foo() {
        return `
    }
    bar() {
        return true;
    }
}
````

I tested the same thing with multi-line strings in c# and the regions stayed collapsed/folded, pointing to a js/ts/x only issue, if this is in fact the same issue as @thany.
I imagine it's potentially tricky to fix this, if it's even considered an 'issue' so to speak, as it's technically valid to create a multi-line template this way and then close it manually a few seconds later.

TL;DR: un-closed template strings will cause all regions after the opening backtick to unfold/expand.

**Comment by ltWolfik** [comment]
Confirm. I have this problem in `.c` files. I fold several functions and after some time suddenly note them expanded. Reproduced already a lot of times, but I don't know the moment exactly when they expanded.

**Comment by aeschli** [comment]
When opening a block comment (`/*`) or a backtick string without the closing token, this will turn a large section of the code into a comment or a string. The parser no longer recognized the structure of the file and doesn't return the folding ranges as before. What was folded in now reopened.

Dup of https://github.com/microsoft/vscode/issues/83353
